### PR TITLE
XML validation infloop followup

### PR DESCRIPTION
**Pull request by jnpkrn**
WIP - missing tests and possibly other corner cases I'll hit

### CODE REVIEW

File: lib/common/schemas.c [hunk 1]
```diff
@@ -867,18 +868,20 @@ cli_config_update(xmlNode **xml, int *best_version, gboolean to_logs)
...
         if (version < min_version) {
-            if (version < orig_version) {
+            if (version < orig_version || orig_version == -1) {
                 if (to_logs) {
-                    crm_config_err("Your current configuration could not validate"
-                                   " with any schema in range [%s, %s], cannot"
-                                   " upgrade to %s.\n",
+                    crm_config_err("Your current configuration %s could not"
+                                   " validate with any schema in range [%s, %s],"
+                                   " cannot upgrade to %s.\n",
```
Comment (kgaillot) [inline lib/common/schemas.c]: Don't need a newline here

**Comment by kgaillot** [comment]
I think the regression tests will be very helpful. Don't worry about the stray newline, I'll get rid of it at some point.

**Comment by jnpkrn** [comment]
Ah, so I fixed pre-existing stray new line and somehow missed the new one (copy-paste, likely) :-)

Didn't bump the title message as I was thinking of adding a doc comment for at least `update_validation` but thsi can come separately as well.

**Comment by kgaillot** [comment]
diff: /usr/share/pacemaker/tests/cli/regression.validity.exp: No such file or directory

Need to edit the makefile to install the .exp

**Comment by jnpkrn** [comment]
Ok, let me send all 3 changes as a separate PR.
